From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Lucia <133892712+plucia-mitre@users.noreply.github.com>
Date: Thu, 19 Dec 2024 14:27:32 -0500
Subject: [PATCH] 7303: Remove an unconstrained context variable in mips16.sinc

Update mips16.sinc

Removing a line which seems to allow the aregs field to be set to any value for encodings of the "restore" instruction, some of which are invalid/reserved.
---
 Ghidra/Processors/MIPS/data/languages/mips16.sinc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Ghidra/Processors/MIPS/data/languages/mips16.sinc b/Ghidra/Processors/MIPS/data/languages/mips16.sinc
index 5977d4decb..0afde1c20e 100644
--- a/Ghidra/Processors/MIPS/data/languages/mips16.sinc
+++ b/Ghidra/Processors/MIPS/data/languages/mips16.sinc
@@ -169,7 +169,6 @@ OFF_M16SP: val(sp) 					is ext_is_ext=0 & m16_iu8_imm & sp [ val = m16_iu8_imm <
 EXT_FRAME: val						is ext_value_frame=0 & m16_svrs_frame=0 [val = 128; ] {export *[const]:2 val;}
 EXT_FRAME: val						is ext_value_frame & m16_svrs_frame [val = ((ext_value_frame << 4) | m16_svrs_frame) << 3;] {export *[const]:2 val;}
 
-REGRS_STAT:							is ext_value_areg {}
 REGRS_STAT:	",a3"					is (ext_value_areg=1 | ext_value_areg=5 | ext_value_areg=9 |ext_value_areg=0xd) {
  	tsp = tsp-4;
 	MemSrcCast(a3,tsp);
-- 
2.45.1

